### ISSUE TITLE
[Bug]: Xcode 14.1 RC + OneSignal purple warnings "This method can cause UI unresponsiveness if invoked on the main thread"

### ISSUE THREAD

**Issue by jdeyrson** [OneSignal/OneSignal-iOS-SDK#1145]
### What happened?

After downloading and running Xcode `Version 14.1 RC 2`, I'm still experiencing this issue https://github.com/OneSignal/OneSignal-iOS-SDK/issues/1113 which generates some purple warnings making the app to stop building as shown in the image:

<img width="1379" alt="Screenshot 2022-10-28 at 2 04 17 PM" src="https://user-images.githubusercontent.com/59494903/198732884-e8b3e005-1e28-4fd1-9227-614d9a74a1b0.png">

### Steps to reproduce?

```Markdown
1. Download the iOS app source code generated at https://gonative.io/ with has OneSignal integration for push notifications.
2. Open GoNativeIOS.xcworkspace in Xcode
3. Build the app pointing to a Simulator device
4. iPhone Simulator goes in blank screen and purple warnings show up
```

### What did you expect to happen?

After building the app, I expect the app to build without errors and run in the simulator.

_Note._ The app successfully runs when it is built pointing out to a **physical iPhone device.** using iOS 15.6.1

### iOS version

16.1

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mehmetdelikaya** [comment]
Hi @jdeyrson 
I have faced with the same warning and tried the suggestion on the below link. It worked for me. You may want to check as well.

https://github.com/OneSignal/OneSignal-iOS-SDK/issues/1113#issuecomment-1248121040

Cheers

**Comment by nan-li** [comment]
Fixed in #1132
